db/commitlog/commitlog.cc

```diff
@@ -1340,6 +1345,9 @@ future<db::commitlog::segment_manager::sseg_ptr> db::commitlog::segment_manager:
                 : make_ready_future<uint64_t>(0)
                 ;
 
+            // if we always overwrite, it should be fine to use this alignment...
```

Comment:
Why is this?

Reply:
According to @avikivity, if the file is pre-written (we only overwrite), we can be allowed to write using the smaller alignment (512 - which should be its own constant - but...). 
Note that the initial write (zeros) still use 4096.

Reply:
This ended up incorrect. We need disk_write_dma_alignment still.

Reply:
How unfortunate. Glö.

Reply:
I'm patching Seastar to supply a lower d_w_d_a when possible.

Reply:
removed.